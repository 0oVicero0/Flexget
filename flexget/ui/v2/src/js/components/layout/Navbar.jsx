import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import IconButton from 'material-ui/IconButton';
import Icon from 'material-ui/Icon';
import Menu, { MenuItem } from 'material-ui/Menu';
import 'font-awesome/css/font-awesome.css';

const styleSheet = createStyleSheet('Navbar', theme => ({
  appBar: {
    position: 'static',
  },
  spacer: {
    flex: 1,
  },
  toolbar: {
    backgroundColor: theme.palette.primary[800],
    height: '100%',
  },
  icon: {
    color: theme.palette.getContrastText(theme.palette.primary[800]),
  },
  menuIcon: {
    paddingRight: 30,
  },
}));

class Navbar extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    toggle: PropTypes.func.isRequired,
    logout: PropTypes.func.isRequired,
  };

  state = {
    open: false,
  };

  handleRequestClose = () => this.setState({ open: false })

  handleMenuClick = event => this.setState({
    open: true,
    anchorEl: event.currentTarget,
  })

  render() {
    const { classes, toggle, logout } = this.props;
    const { anchorEl, open } = this.state;

    return (
      <AppBar className={classes.appBar}>
        <Toolbar className={classes.toolbar}>
          <IconButton
            className={classes.icon}
            onClick={toggle}
          >
            <Icon className="fa fa-bars" />
          </IconButton>
          <div className={classes.spacer} />
          <Link to="/config">
            <IconButton
              className={classes.icon}
              aria-label="Config"
            >
              <Icon className="fa fa-pencil" />
            </IconButton>
          </Link>
          <Link to="/log">
            <IconButton
              className={classes.icon}
              aria-label="Log"
            >
              <Icon className="fa fa-file-text-o" />
            </IconButton>
          </Link>
          <IconButton
            className={classes.icon}
            aria-label="Manage"
            onClick={this.handleMenuClick}
          >
            <Icon className="fa fa-cog" />
          </IconButton>
          <Menu
            id="nav-menu"
            anchorEl={anchorEl}
            open={open}
            onRequestClose={this.handleRequestClose}
          >
            <MenuItem>
              <Icon className={`${classes.menuIcon} fa fa-refresh`} />
              Reload
            </MenuItem>
            <MenuItem>
              <Icon className={`${classes.menuIcon} fa fa-power-off`} />
              Shutdown
            </MenuItem>
            <MenuItem>
              <Icon className={`${classes.menuIcon} fa fa-database`} />
              Database
            </MenuItem>
            <MenuItem onClick={logout}>
              <Icon className={`${classes.menuIcon} fa fa-sign-out`} />
              Logout
            </MenuItem>
          </Menu>
        </Toolbar>
      </AppBar>
    );
  }
}

export default withStyles(styleSheet)(Navbar);
