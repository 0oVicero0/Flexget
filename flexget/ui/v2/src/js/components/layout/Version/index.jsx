import React, { Component } from 'react';
import PropTypes from 'prop-types';
import semver from 'semver-compare';
import IconButton from 'material-ui/IconButton';
import Icon from 'material-ui/Icon';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import 'font-awesome/css/font-awesome.css';

const styleSheet = createStyleSheet('Version', theme => ({
  version: {
    color: theme.palette.error[500],
  },
  par: {
    margin: 0,
  },
}));

class Version extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    version: PropTypes.shape({
      api: PropTypes.string,
      flexget: PropTypes.string,
      latest: PropTypes.string,
    }).isRequired,
    getVersion: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.getVersion();
  }

  render() {
    const { classes, version: { api, flexget, latest } } = this.props;
    return (
      <div className={classes.version}>
        <p className={classes.par}>Version Info</p>
        <p className={classes.par}>Flexget: { flexget } {
          latest && semver(latest, flexget) === 1 && (
            <IconButton href="https://flexget.com/ChangeLog">
              <Icon className={`fa fa-question-circle-o ${classes.version}`} />
            </IconButton>
          ) } </p>
        <p className={classes.par}>API: { api }</p>
      </div>
    );
  }
}

export default withStyles(styleSheet)(Version);
