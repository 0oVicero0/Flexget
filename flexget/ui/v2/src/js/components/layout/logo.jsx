import React from 'react';
import PropTypes from 'prop-types';
import headerImage from 'images/header.png';
import { withStyles, createStyleSheet } from 'material-ui/styles';

const styleSheet = createStyleSheet('Logo', theme => ({
  logo: {
    background: `${theme.palette.accent[900]} url(${headerImage}) no-repeat center`,
    backgroundSize: '175px',
    transition: 'width 0.5s ease',
    height: '100%',
    width: '100%',
  },
}));

const Logo = ({ classes }) => <div className={classes.logo} />;

Logo.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styleSheet)(Logo);
