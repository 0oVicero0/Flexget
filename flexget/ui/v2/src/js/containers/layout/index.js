import { connect } from 'react-redux';
import { checkLogin } from 'actions/auth';
import Layout from 'components/layout';

export function mapStateToProps({ auth }) {
  return {
    loggedIn: !!auth.loggedIn,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    checkLogin: () => dispatch(checkLogin()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Layout);
