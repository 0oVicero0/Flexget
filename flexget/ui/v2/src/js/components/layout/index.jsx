import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import Logo from 'components/layout/Logo';
import Navbar from 'containers/layout/Navbar';
import SideNav from 'components/layout/Sidenav';

const HEADER_HEIGHT = 50;
const MOBILE_HEADER_HEIGHT = (HEADER_HEIGHT * 2) - 2;
const PADDING = 10;

const scrollMixin = theme => ({
  overflowY: 'auto',
  height: `calc(100vh - ${MOBILE_HEADER_HEIGHT}px)`,
  [theme.breakpoints.up('sm')]: {
    height: `calc(100vh - ${HEADER_HEIGHT}px)`,
  },
});
const styleSheet = createStyleSheet('Layout', theme => ({
  layout: {
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
    minHeight: '100vh',
  },
  main: {
    display: 'flex',
    flexDirection: 'row',
    marginTop: MOBILE_HEADER_HEIGHT,
    flex: 1,
    [theme.breakpoints.up('sm')]: {
      marginTop: HEADER_HEIGHT,
    },
  },
  header: {
    display: 'flex',
    minHeight: HEADER_HEIGHT,
    flexDirection: 'column',
    zIndex: 2,
    position: 'fixed',
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      flexDirection: 'row',
    },
  },
  logo: {
    height: HEADER_HEIGHT,
  },
  navbar: {
    height: HEADER_HEIGHT,
    flex: 1,
  },
  sidebar: {
    ...scrollMixin(theme),
  },
  content: {
    flex: 1,
    ...scrollMixin(theme),
  },
  pageWrap: {
    padding: PADDING,
  },
}));

class Layout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    classes: PropTypes.object.isRequired,
    loggedIn: PropTypes.bool,
    checkLogin: PropTypes.func.isRequired,
  };

  static defaultProps = {
    loggedIn: false,
  };

  state = {
    sideBarOpen: false,
  };

  componentDidMount() {
    const { loggedIn, checkLogin } = this.props;
    if (!loggedIn) {
      checkLogin();
    }
  }

  toggleSideBar = () => {
    this.setState({
      sideBarOpen: !this.state.sideBarOpen,
    });
  }

  render() {
    const { children, classes } = this.props;
    const { sideBarOpen } = this.state;

    return (
      <div className={classes.layout}>
        <header className={classes.header}>
          <div className={classes.logo}>
            <Logo sideBarOpen={sideBarOpen} />
          </div>
          <nav className={classes.navbar}>
            <Navbar toggle={this.toggleSideBar} />
          </nav>
        </header>
        <main className={classes.main}>
          <aside className={classes.sidebar}>
            <SideNav sideBarOpen={sideBarOpen} />
          </aside>
          <section className={classes.content}>
            <div className={classes.pageWrap}>
              { children }
            </div>
          </section>
        </main>
      </div>
    );
  }
}

export default withStyles(styleSheet)(Layout);
