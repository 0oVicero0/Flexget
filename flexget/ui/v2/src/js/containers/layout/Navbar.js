import { connect } from 'react-redux';
import { logout } from 'actions/auth';
import Navbar from 'components/layout/Navbar';

function mapDispatchToProps(dispatch) {
  return {
    logout: () => dispatch(logout()),
  };
}

export default connect(null, mapDispatchToProps)(Navbar);

