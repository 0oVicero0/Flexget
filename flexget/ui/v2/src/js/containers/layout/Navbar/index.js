import { connect } from 'react-redux';
import { LOGOUT } from 'actions/auth';
import { SERVER_RELOAD, SERVER_SHUTDOWN } from 'actions/server';
import { request } from 'utils/actions';
import Navbar from 'components/layout/Navbar';

function mapDispatchToProps(dispatch) {
  return {
    logout: () => dispatch(request(LOGOUT)),
    reloadServer: () => dispatch(request(SERVER_RELOAD)),
    shutdownServer: () => dispatch(request(SERVER_SHUTDOWN)),
  };
}

export default connect(null, mapDispatchToProps)(Navbar);

