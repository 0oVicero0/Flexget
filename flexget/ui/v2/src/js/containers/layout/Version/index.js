import { connect } from 'react-redux';
import { getVersion } from 'actions/version';
import Version from 'components/layout/Version';

export function mapStateToProps({ version }) {
  return {
    version,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getVersion: () => dispatch(getVersion()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Version);
