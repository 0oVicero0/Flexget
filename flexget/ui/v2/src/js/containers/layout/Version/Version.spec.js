import { mapStateToProps } from 'containers/layout/Version';

describe('containers/layout/Version', () => {
  it('should return the version', () => {
    expect(mapStateToProps({ version: {
      api: '1.1.2',
      flexget: '2.10.11',
      latest: '2.10.60',
    } })).toMatchSnapshot();
  });
});
