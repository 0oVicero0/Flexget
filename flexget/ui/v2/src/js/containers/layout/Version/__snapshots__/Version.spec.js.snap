// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/layout/Version should return the version 1`] = `
Object {
  "version": Object {
    "api": "1.1.2",
    "flexget": "2.10.11",
    "latest": "2.10.60",
  },
}
`;
